orderapi/src/test/java/fi/hel/verkkokauppa/order/service/invoice/InvoiceServiceUnitTest.java

```diff
@@ -83,22 +84,25 @@ void saveInvoiceToOrder() throws NoSuchMethodException {
                 .ovtId("test-ovtId")
                 .build();
 
-        // Mocking the `save` method of the `OrderRepository` to return the same object that was passed to it.
+        // Mocking the `save` method of the `OrderRepository` to return the same object that was passed to it
+        // + with added generated invoiceId
         Mockito.when(orderRepository.save(any(Order.class))).thenAnswer(i -> {
             Order argument = (Order) i.getArguments()[0];
+            invoiceDto.setInvoiceId(invoiceService.generateOrReapplyInvoiceId(argument));
             argument.setInvoice(invoiceMapper.fromDto(invoiceDto));
             return argument;
         });
 
 
         Order createdOrderWithInvoice = invoiceService.saveInvoiceToOrder(invoiceDto, order);
 
-        Assertions.assertEquals(createdOrderWithInvoice.getInvoice().getBusinessId(), "test-businessId");
-        Assertions.assertEquals(createdOrderWithInvoice.getInvoice().getName(), "test-name");
-        Assertions.assertEquals(createdOrderWithInvoice.getInvoice().getAddress(), "test-address");
-        Assertions.assertEquals(createdOrderWithInvoice.getInvoice().getPostcode(), "test-postcode");
-        Assertions.assertEquals(createdOrderWithInvoice.getInvoice().getCity(), "test-city");
-        Assertions.assertEquals(createdOrderWithInvoice.getInvoice().getOvtId(), "test-ovtId");
+        Assertions.assertEquals("2000000001", createdOrderWithInvoice.getInvoice().getInvoiceId());
```

Comment:
What happens to this assert, if there is more than one invoice? Should this just assert the length of string to be 10 and start with '2'